vim/autoload/himalaya/shared/cli.vim

```diff
@@ -14,8 +14,8 @@ function! himalaya#shared#cli#call(cmd, args, log)
       redraw | call himalaya#shared#log#info(printf("%s [ERR]", a:log))
       for line in split(res, "\n")
         call himalaya#shared#log#err(line)
+        throw line
```

Comment:
The CLI returns a list a errors (a kind of error chain) separated by `\n`. By throwing there we will get only the first error, which is not good. In fact we want this piece of code to throw only when it's called from the telescope previewer, right? Maybe we can pass a new parameter to the function `himalaya#shared#cli#call`, sth like `should_throw` or `throwable`. Feel free to propose a better idea if you have!

Reply:
I was thinking of that before but wasn't sure if the function should throw an error as it did before as well? It earlier did `throw ""` after the for-loop. I now added that back and made it `throw res` instead which we can capture in the preview.

Reply:
What I had in mind is more sth like:

```viml
function! himalaya#shared#cli#call(cmd, args, log, should_throw)
...

for line in split(res, "\n")
  call himalaya#shared#log#err(line)
endfor

if a:should_throw
  throw ""
endif
```

This way, only the `list_preview` will throw:

```viml
function! himalaya#msg#list_preview(mbox, page)
  let msgs = s:cli("--mailbox %s list --page %d", [shellescape(a:mbox), a:page], printf("Fetching %s messages", a:mbox), 1)
```

Reply:
Added the `should_throw` argument 👍